project/src/test/java/com/google/sps/EventSpannerTasksTest.java

```diff
@@ -0,0 +1,121 @@
+package com.google.sps;
+import com.google.cloud.Date;
+import com.google.gson.Gson;
+import com.google.sps.data.Event;
+import com.google.sps.data.User;
+import com.google.sps.data.VolunteeringOpportunity;
+import com.google.sps.servlets.EventCreationServlet;
+import com.google.sps.utilities.SpannerClient;
+import com.google.sps.utilities.SpannerTasks;
+import com.google.sps.utilities.SpannerTestTasks;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.servlet.ServletException;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.springframework.mock.web.MockServletContext;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+/** Unit tests for DatabaseWrapper functionality related to Event class. */
+@RunWith(JUnit4.class)
+public class EventSpannerTasksTest {
+  private static final String HOST_NAME = "Bob Smith";
+  private static final String EMAIL = "bobsmith@example.com";
+  private static final User HOST = new User.Builder(HOST_NAME, EMAIL).build();
+  private static final String EVENT_NAME = "Team Meeting";
+  private static final String NEW_EVENT_NAME = "Daily Team Meeting";
+  private static final String DESCRIPTION = "Daily Team Sync";
+  private static final Set<String> LABELS =
+      Collections.unmodifiableSet(new HashSet<>(Arrays.asList("Tech", "Work")));
+  private static final String LOCATION = "Remote";
+  private static final Date DATE = Date.fromYearMonthDay(2016, 9, 15);
+  private static final String DATE_STRING = "09/15/2016";
+  private static final String TIME = "3:00PM-5:00PM";
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    // Mock a request to trigger the SpannerClient setup to run
+    MockServletContext mockServletContext = new MockServletContext();
+    new SpannerClient().contextInitialized(new ServletContextEvent(mockServletContext));
+    SpannerTestTasks.setup();
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    SpannerTestTasks.cleanup();
+  }
+
+
+  /** Verify insertion of event in db and retrieval by id 
+   * Also tests behavior of EventCreationServlet doGet() where doGet request.getParameter("eventId") == event.getId() */
```

Comment:
Actually, I will write a new test for the doGet() because although it does just get an event back from the db by ID, I want to create the JSON and compare. ( ignore comment on line 63)

Reply:
SG